lib/SetAPI/assembly/AssemblySetInterfaceV1.py

```diff
@@ -1,82 +1,138 @@
+from SetAPI.error_messages import (
+    data_required,
+    include_params_valid,
+    items_list_required,
+    ref_must_be_valid,
+    ref_path_must_be_valid,
+    ref_required,
+)
+from SetAPI.generic.constants import INC_ITEM_INFO, INC_ITEM_REF_PATHS, REF_PATH_TO_SET
 from SetAPI.generic.SetInterfaceV1 import SetInterfaceV1
-from SetAPI.util import info_to_ref
+from SetAPI.util import (
+    check_reference,
+    info_to_ref,
+)
+from typing import Any
 
 
 class AssemblySetInterfaceV1:
-    def __init__(self, workspace_client):
+    def __init__(self: "AssemblySetInterfaceV1", workspace_client):
         self.ws = workspace_client
         self.set_interface = SetInterfaceV1(workspace_client)
 
-    def save_assembly_set(self, ctx, params):
-        if "data" in params:
-            self._validate_assembly_set_data(params["data"])
-        else:
-            raise ValueError('"data" parameter field required to save an AssemblySet')
+    @staticmethod
+    def set_type() -> str:
+        return "KBaseSets.AssemblySet"
+
+    @staticmethod
+    def set_items_type() -> str:
+        return "Assembly"
+
+    @staticmethod
+    def allows_empty_set() -> bool:
+        return True
```

Comment:
Add in some methods so that it's possible to interrogate what kind of sets the module saves and whether or not the module allows the creation of empty sets.

(I'm not convinced that allowing empty sets is deliberate -- looks more like a QC issue)